chatbot-core/backend/app/databases/mssql.py

```diff
@@ -1,66 +1,67 @@
 from sqlalchemy import create_engine
+from sqlalchemy.engine import Engine
 from sqlalchemy.orm import sessionmaker, Session
+from typing import Generator
 
 from app.utils.logger import LoggerFactory
 from app.settings import Constants, Secrets
 
 logger = LoggerFactory().get_logger(__name__)
 
 
-class PostgresConnector:
+class MSSQLConnector:
     """
+    MSSQL connector class
+
     Pattern: Singleton
     Purpose: Create a single instance of the database connection
     """
 
-    _instance = None
-
-    def __init__(self):
-        if PostgresConnector._instance is None:
-            PostgresConnector._instance = self.__create_engine()
+    _instance: Engine = None
 
     @classmethod
-    def get_instance(cls):
+    def get_instance(cls) -> Engine:
         """
         Get the instance of the database connection
         """
         if cls._instance is None:
-            cls._instance = cls().__create_engine()
+            cls._instance = cls()._create_engine()
+
         return cls._instance
 
     @classmethod
-    def __create_engine(cls):
+    def _create_engine(cls) -> Engine | None:
         """
         Create the database connection if there is no any existing connection
         """
         try:
-            uri = Constants.POSTGRES_CONNECTOR_URI.format(
-                user=Secrets.POSTGRES_USER,
-                password=Secrets.POSTGRES_PASSWORD,
-                host=Secrets.POSTGRES_HOST,
-                port=Secrets.POSTGRES_PORT,
-                name=Secrets.POSTGRES_NAME,
+            uri = Constants.MSSQL_CONNECTOR_URI.format(
+                user=Secrets.MSSQL_USER,
+                password=Secrets.MSSQL_SA_PASSWORD,
+                host=Secrets.MSSQL_HOST,
+                db_name=Secrets.MSSQL_DB,
+                driver=Constants.MSSQL_DRIVER,
             )
             return create_engine(
                 uri,
-                pool_size=Constants.POSTGRES_POOL_SIZE,
-                max_overflow=Constants.POSTGRES_MAX_OVERFLOW,
-                pool_timeout=Constants.POSTGRES_POOL_TIMEOUT,
-                pool_recycle=Constants.POSTGRES_POOL_RECYCLE,
+                pool_size=Constants.MSSQL_POOL_SIZE,
+                max_overflow=Constants.MSSQL_MAX_OVERFLOW,
+                pool_timeout=Constants.MSSQL_POOL_TIMEOUT,
+                pool_recycle=Constants.MSSQL_POOL_RECYCLE,
             )
         except Exception as e:
             logger.error(f"Error initializing database: {e}")
             raise
 
 
-# Get the engine from the PostgresConnector singleton
-engine = PostgresConnector.get_instance()
+# Get the engine from the MSSQLConnector singleton
+engine = MSSQLConnector.get_instance()
 
 # Create a session maker
 SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
 
 
-def get_session() -> Session:
+def get_db_session() -> Generator[Session, None, None]:
```

Comment:
The function `get_db_session` is a parameter of the FastAPI `Depends` function for dependency injection. `Depends` cannot receive the function which is wrapped by `contextmanager`.

Reply:
Checkout my new commit [ced6424](https://github.com/TinhHoaSolutions-EzHR/chatbot/pull/50/commits/ced64248dbf812d9a9f29cb8785995dbef36de70).

Cannot wrap this function by `contextmanager` because it will cause this error `TypeError: db_session must be an instance of Session, got <class 'contextlib._GeneratorContextManager'>`. And I prefer not adding this block code

```python
except Exception:
    session.rollback()
    raise
```

as it will cause double raise error (I am sorry because I forgot what this error looks like but sure that we don't need this block of code :D).